src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -1393,4 +1401,80 @@ export class NetworkAreaDiagramViewer {
             }
         });
     }
+
+    public setJsonBranchStates(branchStates: string) {
+        const branchStatesArray: BranchState[] = JSON.parse(branchStates);
+        this.setBranchStates(branchStatesArray);
+    }
+
+    public setBranchStates(branchStates: BranchState[]) {
+        branchStates.forEach((branchState) => {
+            if (!this.edgesMap.has(branchState.branchId)) {
+                const edge: EdgeMetadata | undefined = this.diagramMetadata?.edges.find(
+                    (edge) => edge.equipmentId == branchState.branchId
+                );
+                if (edge === undefined) {
+                    console.warn('Skipping updating branch ' + branchState.branchId + ' labels: branch not found');
+                    return;
+                }
+                this.edgesMap.set(branchState.branchId, edge);
+            }
+            this.setBranchSideLabel(
+                branchState.branchId,
+                '1',
+                this.edgesMap.get(branchState.branchId)?.svgId ?? '-1',
+                branchState.value1
+            );
+            this.setBranchSideLabel(
+                branchState.branchId,
+                '2',
+                this.edgesMap.get(branchState.branchId)?.svgId ?? '-1',
+                branchState.value2
+            );
+            this.setBranchSideConnection(
+                branchState.branchId,
+                '1',
+                this.edgesMap.get(branchState.branchId)?.svgId ?? '-1',
+                branchState.connected1
+            );
+            this.setBranchSideConnection(
+                branchState.branchId,
+                '2',
+                this.edgesMap.get(branchState.branchId)?.svgId ?? '-1',
+                branchState.connected2
+            );
+        });
+    }
+
+    private setBranchSideLabel(branchId: string, side: string, edgeId: string, value: number | string) {
+        const halfEdge: SVGGraphicsElement | null = this.container.querySelector("[id='" + edgeId + '.' + side + "']");
+        const arrowGElement = halfEdge?.lastElementChild?.firstElementChild;
```

Comment:
and here
```typescript
const arrowGElement: SVGGraphicsElement | null = this.container.querySelector("[id='" + edgeId + '.' + side + "'] .nad-edge-infos g");
```

Note that arrowGElement cannot be undefined then, so no need to check this.

Reply:
querySelector used, instead of getting child